OptimizelySDK/Optimizely.cs

```diff
@@ -308,5 +296,226 @@ public Variation GetForcedVariation(string experimentKey, string userId)
+            
+            if (variableValue != null && Boolean.TryParse(variableValue, out bool booleanValue))
+                return booleanValue;
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets double feature variable value.
+        /// </summary>
+        /// <param name="featureKey">The feature flag key</param>
+        /// <param name="variableKey">The variable key</param>
+        /// <param name="userId">The user ID</param>
+        /// <param name="userAttributes">The user's attributes.</param>
+        /// <returns>double | Feature variable value or null</returns>
+        public double? GetFeatureVariableDouble(string featureKey, string variableKey, string userId, UserAttributes userAttributes)
+        {
+            var variableValue = GetFeatureVariableValueForType(featureKey, variableKey, userId,
+                userAttributes, FeatureVariable.VariableType.DOUBLE);
+            
+            if (variableValue != null && Double.TryParse(variableValue, out double doubleVaue))
+                return doubleVaue;
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets integer feature variable value.
+        /// </summary>
+        /// <param name="featureKey">The feature flag key</param>
+        /// <param name="variableKey">The variable key</param>
+        /// <param name="userId">The user ID</param>
+        /// <param name="userAttributes">The user's attributes.</param>
+        /// <returns>int | Feature variable value or null</returns>
+        public int? GetFeatureVariableInteger(string featureKey, string variableKey, string userId, UserAttributes userAttributes)
+        {
+            var variableValue = GetFeatureVariableValueForType(featureKey, variableKey, userId,
+                userAttributes, FeatureVariable.VariableType.INTEGER);
+            
+            if (variableValue != null && Int32.TryParse(variableValue, out int integerValue))
+                return integerValue;
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets string feature variable value.
+        /// </summary>
+        /// <param name="featureKey">The feature flag key</param>
+        /// <param name="variableKey">The variable key</param>
+        /// <param name="userId">The user ID</param>
+        /// <param name="userAttributes">The user's attributes.</param>
+        /// <returns>string | Feature variable value or null</returns>
+        public string GetFeatureVariableString(string featureKey, string variableKey, string userId, UserAttributes userAttributes)
+        {
+            return GetFeatureVariableValueForType(featureKey, variableKey, userId, userAttributes, 
+                FeatureVariable.VariableType.STRING);
+        }
+
+        /// <summary>
+        /// Sends impression event.
+        /// Send an impression event if the user is bucketed into an experiment using the feature.
```

Comment:
Remove this. This method does not know this. It gets called both from isFeatureEnabled and activate

Reply:
Done